docs/Offline-Signatures-QR-Code.md

```diff
@@ -0,0 +1,280 @@
+# Off-line Signature QR Code
+
+## Table of Contents
+
+- [Introduction](#introduction)
+- [Operation Attributes](#operation-attributes)
+- [Operation Data](#operation-data)
+- [Flags](#flags)
+- [Signature Calculation](#signature)
+- [Forward Compatibility](#forward-compatibility)
+- [Offline Signature Process Description](#offline-signature-process-description)
+
+## Introduction
+
+This chapter describes how operation data is encoded into QR code for the purpose of off-line verification.
+
+## Operation Attributes
+
+The format of QR code for offline operations is a high density string composed from several main operation attributes:
+
+1. `{OPERATION_ID}` - operation identifier (UUID like identifier)
+2. `{TITLE}` - title for operation, in UTF8 format. For example: "Payment".
+   - ASCII control characters (code < 32) are forbidden
+   - `\n` can be used for newline character
+   - `\\` can be used for backslash
+3. `{MESSAGE}` - message associated with operation, in UTF8 format. For example: "Please confirm this playment"
+   - ASCII control characters (code < 32) are forbidden
+   - `\n` can be used for newline character
+   - `\\` can be used for backslash
+4. `{OPERATION_DATA}` - content of operation data
+5. `{FLAGS}` - various flags affecting how operation is processed
+6. `{NONCE_B64}` - nonce, 16 random bytes in Base64 format. Required for pure offline signature.
+7. `{SIGNING_KEY_TYPE}{ECDSA_QRDATA_SIGNATURE_BASE64}` - ECDSA signature in Base64 format. The signature string has an one prefix character, containing an information about signing key type.
+
+Then, the final string for QR code is simple, new-line separated list of attributes:
+
+```
+{OPERATION_ID}\n
+{TITLE}\n
+{MESSAGE}\n
+{OPERATION_DATA}\n
+{FLAGS}\n
+{NONCE_B64}\n
+{SIGNING_KEY_TYPE}{ECDSA_QRDATA_SIGNATURE_BASE64}
+```
+
+For example:
+```
+5ff1b1ed-a3cc-45a3-8ab0-ed60950312b6
+Payment
+Please confirm this payment
+A1*A100CZK*ICZ2730300000001165254011*D20180425
+B
+AD8bOO0Df73kNaIGb3Vmpg==
+0MEYCIQDby1Uq+MaxiAAGzKmE/McHzNOUrvAP2qqGBvSgcdtyjgIhAMo1sgqNa1pPZTFBhhKvCKFLGDuHuTTYexdmHFjUUIJW=
+```
+
+## Operation Data
+
+Operation data is an asterisk separated list of fields, where the first field defines a version of operation data and template. Other fields are additional and contains typically significant attributes, which has to be presented to the user, before the operation is confirmed and signed.
+
+An exact details of the operation data structure can be found in the separate documentation:
+
+- [Operation Data Structure](./Operation-Data.md)
+
+## Flags
+
+`{FLAGS}` attribute is a string of characters, where each character represents one flag. Order of characters in string is not important.
+
+| Flag | Meaning                      |
+|------|:-----------------------------|
+| `B`  | Operation can be signed with biometric factor |
```

Comment:
If QR code documentation will be part of enrollment documentation, we should also document new flags introduced in 1.4.0; It may be another issue and PR.

Reply:
I think I will add it for now, just for the sake of having all documentation related to operation data (online / offline) and operation form data in one place. We can revise this later, if needed.